### PR TITLE
Default to building node-gyp dependencies

### PR DESCRIPTION
**Pull request by bonkydog**
npm v6 defaults to building node-gyp dependencies lacking install or preinstall scripts.

It accomplishes this by adding an install script to `package.json`s as they are read.

Here we are moving this logic closer to where we decide to rebuild a dependency.

## References
https://github.com/npm/cli/issues/1848

### CODE REVIEW

File: lib/arborist/rebuild.js [hunk 1]
```diff
@@ -178,6 +179,10 @@ module.exports = cls => class Builder extends cls {
       if (bin)
         await this[_checkBins](node)
       set.add(node)
+    } else if (!install && !preinstall && await isNodeGypPackage(node.path)) {
```
Comment (isaacs) [inline lib/arborist/rebuild.js:182]: Hmm, this skips the `_checkBins` if it's a nodegyp package, and skips the node-gyp default if there's bins.

Could do something like this?

```js
const isGyp = !install && !preinstall && await isNodeGypPackage(node.path)
if (bin || preinstall || install || postinstall || isGyp) {
  if (bin)
    await this[_checkBins](node)
  if (isGyp)
    node.package.scripts.install = 'node-gyp rebuild'
  set.add(node)
}
```

**Review by isaacs — CHANGES_REQUESTED** [review]
Edge case where bins aren't being checked in the "isNodeGyp" case, and node-gyp-ness isn't checked if there are bins.

Other than that, lgtm.

**Comment by isaacs** [comment]
Also, the test fixture being added here is rather large and unwieldy... maybe we could simulate node-report instead of including something that actually creates a binary?

File: lib/arborist/rebuild.js (hunk 1 above)
Comment (bonkydog) [reply lib/arborist/rebuild.js:182]: Yeah, looks good.  I'll do it!

Comment (bonkydog) [reply lib/arborist/rebuild.js:182]: Done.

**Comment by bonkydog** [comment]
> Also, the test fixture being added here is rather large and unwieldy... maybe we could simulate node-report instead of including something that actually creates a binary?

I swapped node-report out for node-gyp-hello-world. It still does a real build, but it's lighter weight.
